main.tf

```
@@ -1,16 +1,140 @@
+module "lambda_package" {
+  source  = "matti/resource/shell"
+  version = "~>1.0.7"
 
-module "tags" {
-  source  = "rhythmictech/tags/terraform"
-  version = "1.0.0"
+  # I want to be using `npm run-script build` here,
+  # but for some reason I can't work out `node-lambda` fails in that context
+  command     = "node-lambda package -e production -x 'event.json build.env bin .github' -I lambci/lambda:build-nodejs12.x"
+  working_dir = "${path.module}/lambda"
 
-  enforce_case = "UPPER"
-  names        = [var.name]
-  tags         = var.tags
+  trigger = join("", values({
+    package_lock_json = filemd5("${path.module}/lambda/package-lock.json")
+    package_json      = filemd5("${path.module}/lambda/package.json")
+    app               = filemd5("${path.module}/lambda/index.js")
+  }))
+}
+
+data "aws_iam_policy_document" "assume" {
+  statement {
+    actions = [
+      "sts:AssumeRole",
+    ]
+
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_role" "this" {
+  name_prefix = "${var.name}-"
```

```suggestion
  name_prefix = "${var.name}"
```
I thought you were pushing us away from the superfluous dash

---

Nope. IMO it isn't superfluous. It creates a visual break between the prefix and the terraform-generated stuff.